### PR TITLE
new ioctl cmd alias set and unit test

### PR DESCRIPTION
**Pull request by RuneAkko**
#1770

### CODE REVIEW

File: ioctl/newcmd/alias/aliasset.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+			for aliases[addr] != "" {
+				delete(c.Config().Aliases, aliases[addr])
+				aliases = GetAliasMap()
+			}
+			c.Config().Aliases[alias] = addr
+			out, err := yaml.Marshal(c.Config())
+			if err != nil {
+				return output.NewError(output.SerializationError, "failed to marshal config", err)
+			}
+			if err := c.WriteFile(config.DefaultConfigFile, out, 0600); err != nil {
```
Comment (yutongp) [inline ioctl/newcmd/alias/aliasset.go:63]: do we support changing config file path? if so, here we shall write to user defined config file as well?

File: ioctl/newcmd/alias/aliasset.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		config.English: "Set alias for address",
+		config.Chinese: "设定地址的别名",
+	}
+	setCmdUses = map[config.Language]string{
+		config.English: "set ALIAS ADDRESS",
+		config.Chinese: "set 别名 地址",
+	}
+)
+
+// NewAliasSet represents the alias set command
```
Comment (CoderZhi) [inline ioctl/newcmd/alias/aliasset.go:33]: wrong comment

File: ioctl/client.go [hunk 3]
```diff
@@ -186,6 +190,10 @@ func (c *client) GetAliasMap() map[string]string {
 	return aliases
 }
 
+func (c *client) WriteFile(filename string, data []byte, perm os.FileMode) error {
+	return ioutil.WriteFile(filename, data, perm)
+}
```
Comment (CoderZhi) [inline ioctl/client.go:195]: For client, the function should be `func (c *client) WriteConfig(cfg Config) error`, the config file path and the file mode should be defined and used inside of `client`

File: ioctl/newcmd/alias/aliasset.go (hunk 1 above)
Comment (RuneAkko) [reply ioctl/newcmd/alias/aliasset.go:63]: yes, some ioctl commands can change config file. The old ioctl command doesn't seem to support user defined config dir. Do I need to add this feature?

Comment (yutongp) [reply ioctl/newcmd/alias/aliasset.go:63]: i think for newcmd, we should use user defined config if that's possible. for legacy ioctl cmd if it doesn't support this feature, it is ok.

File: ioctl/client.go (hunk 3 above)
Comment (RuneAkko) [reply ioctl/client.go:195]: changed.
Now the "cfg Config" is from cmd/config.go, do I need to migrate config.go?

File: ioctl/newcmd/alias/aliasset.go (hunk 1 above)
Comment (RuneAkko) [reply ioctl/newcmd/alias/aliasset.go]: sure. I will add a new config cmd flag about custom dir. Maybe I can do it when migrate cmd/config.go cmd?

**Review by CoderZhi — APPROVED** [review]

**Comment by millken** [comment]
can be closed

**Pull request merged**